File: fix_urls.js
const fs = require('fs');
const path = require('path');

const dir = 'D:\\SAMCAFFEE\\frontend\\src\\services';
const files = fs.readdirSync(dir);

files.forEach(file => {
    if (!file.endsWith('.ts')) return;
    const filepath = path.join(dir, file);
    let content = fs.readFileSync(filepath, 'utf8');
    
    const regex = /const API_URL = ([^;]+);/;
    const match = content.match(regex);
    if (match) {
        const oldVal = match[1];
        let endpoint = '';
        if (file.toLowerCase().includes('auth')) endpoint = '/auth';
        else if (file.toLowerCase().includes('category')) endpoint = '/categories';
        else if (file.toLowerCase().includes('ingredient')) endpoint = '/ingredients';
        else if (file.toLowerCase().includes('order')) endpoint = '/orders';
        else if (file.toLowerCase().includes('payment')) endpoint = '/payments';
        else if (file.toLowerCase().includes('product')) endpoint = '/products';
        else if (file.toLowerCase().includes('shift')) endpoint = '/shifts';
        else if (file.toLowerCase().includes('stat') || file.toLowerCase().includes('analytic') || file.toLowerCase().includes('dashboard')) endpoint = '/stats';
        else if (file.toLowerCase().includes('user')) endpoint = '/users';
        
        const newVal = '`${process.env.NEXT_PUBLIC_API_URL || "http://localhost:3001"}' + endpoint + '`';
        content = content.replace(`const API_URL = ${oldVal};`, `const API_URL = ${newVal};`);
        fs.writeFileSync(filepath, content, 'utf8');
        console.log(`Fixed ${file} with endpoint ${endpoint}`);
    }
});

File: fix_urls.py
import os, re
path = r'D:\SAMCAFFEE\frontend\src\services'
for file in os.listdir(path):
    if not file.endswith('.ts'): continue
    filepath = os.path.join(path, file)
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    match = re.search(r'const API_URL = ([^;]+);', content)
    if match:
        old_val = match.group(1)
        endpoint = ''
        if 'auth' in file.lower(): endpoint = '/auth'
        elif 'category' in file.lower(): endpoint = '/categories'
        elif 'ingredient' in file.lower(): endpoint = '/ingredients'
        elif 'order' in file.lower(): endpoint = '/orders'
        elif 'payment' in file.lower(): endpoint = '/payments'
        elif 'product' in file.lower(): endpoint = '/products'
        elif 'shift' in file.lower(): endpoint = '/shifts'
        elif 'stat' in file.lower() or 'analytic' in file.lower() or 'dashboard' in file.lower(): endpoint = '/stats'
        elif 'user' in file.lower(): endpoint = '/users'
        
        new_val = f'`${{process.env.NEXT_PUBLIC_API_URL || "http://localhost:3001"}}{endpoint}`'
        content = content.replace(f'const API_URL = {old_val};', f'const API_URL = {new_val};')

        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f"Fixed {file} with endpoint {endpoint}")
